docs/extend/components.md

```diff
@@ -109,8 +111,386 @@ const AnotherComponent = Ractive.extend({
+
+While not as flexible as unmanaged components, the managed approach clearly communicates all the available components through it's template.
+
+With managed components we just need an anchor and can attach any Ractive instance to it:
+```html
+<#child/>
+```
+
+```ja
+parent.attachChild( compA, { target: "child" } );
+
+// Later on
+parent.detachChild( compA );
+parent.attachChild( childB, { target: "child" } );
+
+// Still later
+parent.detachChild( compB );
+parent.attachChild( childC, { target: "child" } );
+```
+
+### Workflow
+Ractive provides a variety of ways that parent and child components can communicate with each other.
+
+Components are great for encapsulating features in their own file and provides a way to reuse compponents in an application, or even across different applications.
+
+For maximum reusability a component should be fully encapsulated and not have any dependencies on it's parent.
+
+To achieve this, communication must be __one way__, from the parent to the child. So the parent can **map data** for the child or invoke the child API but the child must not access the parent instance in order call methods etc. For example, if the child wants to communicate a change to it's parent and does so by calling a parent method, then wherever the child component is used in the application, the parent is responsible for providing the method the child depends upon. So if the child makes 10 calls to the parent, all 10 methods must be made available to the child, which makes reuse difficult. Ideally the child should use a different mechanism to communicate to it's parent where it does not require access the parent.
+
+The solution is for the child to communicate to it's parent, by firing events, that the parent can listen to. This way the child has no dependency on it's parent. The parent can decide which events it is interested in and how to respond to them. Generally the parent will communicate with the child through the child' API. When the parent receives an event from the child, it can respond by calling methods on the child instance.
+
+This is illustrated in the following diagram:
+
+![one-way-data-flow](../static/img/component-one-way-data-flow.png)
+
+Breaking the dependency between components is great for reusability, but comes at a cost since firing and responding to events is more involved than simply calling methods on the parent.
+
+Let's look at the various ways parent and child components can communicate
+
+#### Data mappings
+
+**Data mapping** is a mechanism that allows us to *connect* pieces of data in the parent to data in the child. When a **data mapping** is specified Ractive creates a [link()](../api/instance-methods.md#ractivelink) between the parent and child data.
+
+Changes on one side will reflect on the other and vice versa.
```

Comment:
With docs I tend to drag everyone to my level, in other words dumb it down ;-)

So how about this (feel free to change it, and I'll update):
**Data mapping** is a mechanism to *expose* a piece of data in the parent to the child. When a **data mapping** is specified Ractive creates a [link()](../api/instance-methods.md#ractivelink) between the parent and child so that the child can *accesss* the parent data through *keypaths*. Changes on one side will reflect on the other and vice versa.

*Note:* specifying a *data mapping* does not create a *copy* of the data in the child. [Linking()](../api/instance-methods.md#ractivelink) simply informs Ractive that two keypaths refer to the same data.